site/js/MessagesActor.js

```diff
@@ -7,9 +7,17 @@ export function queryByDataset(key, value) {
 
 export default class MessagesActor extends Actor {
   init() {
-    this.actors.channels.on('update active channel', () => {
+    this.actors.channels.on('update active channel', async channel => {
       this.clear()
-      // TODO fetch recent messages
+      
+      // Display latest messages in the channel
+      const { messages } = await get(`channel/${channel.id}/latest-messages`)
+      for (const msg of messages) {
+        await this.showMessage(msg)
+      }
+
+      const getScrollDist = () => 
+      messagesContainer.scrollTop = messages.scrollHeight - messages.offsetHeight
```

Comment:
Not sure what this is for? It looks like you were part-way through copy/pasting the code for scrolling to the (new) bottom after appending messages, but I think that logic isn't necessary since you'll just be scrolling to the bottom anyways.

..And the lone `const getScrollDist = () =>` line means that the following line might not being executed (since it thinks that line is part of `getScrollDist`)? That would be bad!

Reply:
oh man